draft-ietf-quic-tls.md

```diff
@@ -1519,6 +1584,52 @@ SHOULD track redundant packets and treat excessive volumes of any non-productive
 packets as indicative of an attack.
 
 
+## Packet Number Protection Analysis {#pn-encrypt-analysis}
+
+Packet number protection relies on the randomness of the AEAD output, which is a
+property that AEAD algorithms do not guarantee.  Therefore, no strong assurances
+about the general security of this mechanism can be proven.
+
+Use of the same key and nonce more than once can weaken the guarantees provided
+by this protection.  For the schemes described, protecting two different packet
+numbers with the same key and nonce reveals the exclusive OR of those packet
+numbers, which might be used to compromise confidentiality.  For packet number
+protection to be effective, the output of the packet protection AEAD needs to be
+effectively random.
+
+All the AEAD functions used meet indistinguishability under (adaptive) chosen
+plaintext attack (IND-CPA, IND-CPA2) goals and produce minimal expansion of the
+plaintext, adding only an authentication tag.  Therefore, this document assumes
+that each bit of sampled AEAD output contains one bit of entropy and that an
+attacker is unable to reduce this without knowledge of the key.  Based on this
+assumption, the odds of two samples of ciphertext being identical approach the
+birthday bound for the size of the sample (that is, two to the negative power of
+half the number of sampled bits).
```

Comment:
Possible simplification of these three paragraphs: 

Assuming all AEAD functions used are reducible to PRFs, an (encoded) packet number p is encrypted as:

~~~
    p XOR PRF(pn_key, m),
~~~

where m is an L-bit string sampled from a protected packet’s ciphertext. This construction is secure 
against chosen plaintext attacks (IND-CPA) if the distribution of m is uniform over {0,1}^L {{KatzLindell}}. 
Since AEAD schemes are semantically secure, we assume each bit of sampled ciphertext contains one bit 
of entropy. Thus, two ciphertexts will collide, possibly leading to nonce reuse, with probability bounded by 
2^(-L/2), i.e., the birthday bound. Use of the same key and nonce more than once can weaken the 
guarantees provided by this protection. For the schemes described, protecting two different packet 
numbers with the same key and nonce reveals the exclusive OR of those packet numbers, which might be 
used to compromise confidentiality. 


KatzLindell: Katz, Jonathan, and Yehuda Lindell. Introduction to modern cryptography. CRC press, 2014.

Reply:
I think that's the right formulation: we assume that the AEAD is a PRF.  It's not guaranteed to be by the contract, but it's probably true for the AEADs here.  Thanks for the text.  I'll probably want to make the assumption much more prominent though.  Also, the general form here is using AES or ChaCha as a PRF, so that means that we can factor out the cipher-specific parts.

I'll see what I can do.